03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,334 @@
+all.equal(colnames(df), metadata$refinebio_accession_code)
+```
+
+## How are our samples clustering together?
+
+Now we are going to use a combination of functions from the `DESeq2` and 
+`pheatmap` packages to look at how are samples and genes are clustering.
+
+### Create a DESeqDataset
+We will be using the `DESeq2` package for normalization, which requires us to
+format our data into a `DESeqDataSet` object. We turn the data.frame (or matrix)
+into a [`DESeqDataSet` object](TODO: Add link to header section) and specify
+which variable labels our experimental groups using the `design` argument. 
+
+```{r}
+# The `DESeqDataSetFromMatrix` function needs the values to be converted to
+# integers
+df <- df %>%
+  # Mutate numeric variables to be integers
+  dplyr::mutate_if(is.numeric, round)
+
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = df, # This is the data.frame with the counts values for all replicates in our dataset
+  colData = metadata, # This is the data.frame with the annotation data for the replicates in the counts data.frame
+  design = ~refinebio_treatment # Replace with an appropriate design variable for your analysis
+)
+```
+
+### Define a minimum counts cutoff
+We want to filter out the genes that have not been expressed or that have
+low expression counts because we want to remove any possible noise from our data
+before we normalize the data and create our heatmap. We are going to do some
+pre-filtering to keep only genes with 10 or more reads total. Note that rows
+represent gene data and the columns represent sample data in our dataset.
+
+```{r}
+# Let's take a look at the distribution of counts first
+
+# Sum up the counts for each gene -- which means we want the sum of counts for
+# each row as the rows represent gene data and the columns represent sample data
+# in our dataset
+count_sums <- data.frame(count = rowSums(counts(dds)))
+
+# Create a density plot using the sums
+ggplot2::ggplot(count_sums, ggplot2::aes(x = count)) +
+  ggplot2::geom_density() +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = 10), # This line represents the current cutoff of 10
+    color = "blue",
+    linetype = "solid"
+  ) +
+  ggplot2::geom_vline(ggplot2::aes(xintercept = mean(count)), # This line represents the mean count value
+    color = "red",
+    linetype = "dashed"
+  )
+```
+
+It appears that our arbitrary cutoff of 10 (the solid blue line) is a decent
+cutoff to make here as it seems to be just about inline with the mean counts
+value (the red dashed line) and retains the data points that are not in the
+majority and therefore may hold significance.
+
+```{r}
+# Define a minimum counts cutoff and filter `DESeqDataSet` object to include
+# only rows that have counts above the cutoff
+genes_to_keep <- rowSums(counts(dds)) >= 10
+dds <- dds[genes_to_keep, ]
+```
+
+### Perform DESeq2 normalization
+We are going to use the `rlog` function from the `DESeq2` package to normalize
```

Comment:
```suggestion
We are going to use the `rlog()` function from the `DESeq2` package to normalize
```

This reminds me. I need to add this to the style guide. I think we want to stick with the notation @jashapiro came up with over on `training-modules` which is when we are talking about functions, always include the `()`. I think this helps users low-key keep track of what things are functions.